net.solarnetwork.node.weather.ibm.wc/src/net/solarnetwork/node/weather/ibm/wc/BasicWCClient.java

```diff
@@ -0,0 +1,269 @@
+
+package net.solarnetwork.node.weather.ibm.wc;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.net.URLConnection;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import org.joda.time.LocalTime;
+import com.fasterxml.jackson.databind.JsonNode;
+import net.solarnetwork.node.domain.GeneralDayDatum;
+import net.solarnetwork.node.support.JsonHttpClientSupport;
+
+public class BasicWCClient extends JsonHttpClientSupport implements WCClient {
+
+	/** The default value for the {@code baseUrl} property. */
+	public static final String DEFAULT_BASE_URL = "https://api.weather.com";
+
+	/** The daily forecast url format */
+	public static final String DEFAULT_DAILY_FORECAST_URL = "v3/wx/forecast/daily/%s";
+
+	/** The default output format */
+	public static final String DEFAULT_FORMAT = "format=json";
+
+	/** The default value for the {@code locationTemplate} property */
+	public static final String DEFAULT_LOCATION_TEMPLATE = "icaoCode=%s";
+
+	/** The hourly forecast url format */
+	public static final String DEFAULT_HOURLY_FORECAST_URL = "v3/wx/forecast/hourly/%s";
+
+	/** The default value for the {@code apiKeyTemplate} property. */
+	public static final String DEFAULT_API_KEY_TEMPLATE = "apiKey=%s";
+
+	/** The default value for the {@code languageTemplate} property. */
+	public static final String DEFAULT_LANGUAGE_TEMPLATE = "language=en-US";
+
+	/** The default value for the {@code units} property. */
+	public static final String DEFAULT_UNITS = "units=e";
+
+	public static final boolean DEFAULT_COMPRESSION = true;
+
+	private String baseUrl;
+	private String dailyForecastUrl;
+	private String hourlyForecastUrl;
+	private String defaultFormat;
+	private String apiKeyTemplate;
+	private String locationTemplate;
+	private String languageTemplate;
+	private String units;
+	private Boolean test;
+
+	public BasicWCClient() {
+		this.baseUrl = DEFAULT_BASE_URL;
+		this.dailyForecastUrl = DEFAULT_DAILY_FORECAST_URL;
+		this.hourlyForecastUrl = DEFAULT_HOURLY_FORECAST_URL;
+		this.defaultFormat = DEFAULT_FORMAT;
+		this.apiKeyTemplate = DEFAULT_API_KEY_TEMPLATE;
+		this.locationTemplate = DEFAULT_LOCATION_TEMPLATE;
+		this.languageTemplate = DEFAULT_LANGUAGE_TEMPLATE;
+		this.units = DEFAULT_UNITS;
+		this.test = false;
+
+		// set default compression
+		this.setCompress(DEFAULT_COMPRESSION);
+	}
+
+	private String getURL(String requestURL, String location, String apiKey, String datumPeriod) {
+		if ( test ) {
+			return this.baseUrl + "/" + String.format(requestURL, datumPeriod);
+		}
```

Comment:
There should be no testing for "unit test mode" here. Instead the class should be generating a valid URL so the unit test can verify the constructed URL.